implementations/micrometer-registry-graphite/src/test/java/io/micrometer/graphite/GraphiteTagNameMapperTest.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.graphite;
+
+import io.micrometer.core.Issue;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link GraphiteTagNameMapper}.
+ *
+ * @author Jon Schneider
+ * @author Johnny Lim
+ * @author Andrew Fitzgerald
+ */
+class GraphiteTagNameMapperTest {
+    private final GraphiteTagNameMapper nameMapper = new GraphiteTagNameMapper("stack", "app.name");
+    private final Meter.Id id = new SimpleMeterRegistry().counter("my.name",
+            "app.name", "MYAPP", "stack", "PROD", "other.tag", "value").getId();
+
+    @Issue("#561")
+    @Test
+    void tagsAsPrefix() {
+        assertThat(nameMapper.toHierarchicalName(id, NamingConvention.camelCase))
+                .isEqualTo("PROD.MYAPP.myName;otherTag=value");
```

Comment:
Does it make sense to allow tag prefixing if actual tags are an option?

Reply:
I would think not? Can you think of a case where we would want it? We could adopt the view here too that it would be peculiar for someone to want some mixture of prefixing and tags support, that if they did they'd likely want to configure one tag as a prefix and the rest as Graphite tags, etc. And the API already supports them creating their own `HierarchicalNameMapper` and doing just that.

Reply:
a1bb940f936c692f8c3411821e0ccbe7b096e775